synadm/api.py

```diff
@@ -28,12 +28,39 @@
 https://matrix.org/docs/spec/#matrix-apis.
 """
 
-import requests
-from http.client import HTTPConnection
 import datetime
 import json
-import urllib.parse
 import re
+from typing import Optional, Union, Dict, List, Any
+import urllib.parse
+from http.client import HTTPConnection
+import traceback
+
+import requests
+from requests.exceptions import ConnectionError
```

Comment:
My linter keeps telling me that this is not required to import since it is built-in. The damn thing is wrong though and you are right. This must be imported otherwise the exception is not catched. (Damnit `pylint` I was counting on you! 🤣 )

Reply:
Yes, that's how we should do it: 4bb093d. Pylint was actually complaining that I overwrite the built-in with `as ConnectionError`. This is also fixed now.